diffusion/models/stable_diffusion.py

```diff
@@ -366,6 +440,23 @@ def generate(
         # scale the initial noise by the standard deviation required by the scheduler
         latents = latents * self.inference_scheduler.init_noise_sigma
 
+        added_cond_kwargs = {}
+        # if using SDXL, prepare added time ids & embeddings
+        if self.sdxl and pooled_embeddings is not None:
+            if not crop_params:
+                crop_params = [0., 0.]
+            if not size_params:
+                size_params = [width, height]
+            cond_original_size = torch.tensor([[width, height]]).repeat(pooled_embeddings.shape[0],
+                                                                        1).to(device).float()
+            cond_crops_coords_top_left = torch.tensor([crop_params]).repeat(pooled_embeddings.shape[0],
+                                                                            1).to(device).float()
+            cond_target_size = torch.tensor([size_params]).repeat(pooled_embeddings.shape[0], 1).to(device).float()
+            add_time_ids = torch.cat([cond_original_size, cond_crops_coords_top_left, cond_target_size], dim=1).float()
```

Comment:
What do you think of (not necessary):
```suggestion
            add_time_ids = torch.tensor([[width, height, *crop_params, *size_params]], dtype=torch.float, device=device)
            add_time_ids = add_time_ids.repeat(pooled_embeddings.shape[0], 1)
```